backend/benefit/users/tests/test_gdpr_api.py

```diff
@@ -0,0 +1,42 @@
+import json
+
+import pytest
+from django.contrib.auth import get_user_model
+from django.test import override_settings
+from django.urls import reverse
+
+from shared.common.tests.factories import UserFactory
+from users.models import format_date
+
+User = get_user_model()
+
+
+def extract_val(resp: dict, key: str) -> str:
+    """Extract value from GDPR API response."""
+    user = resp["children"]
+    value_pair = next(item for item in user if item["key"] == key)
+    return value_pair["value"]
+
+
+@override_settings(NEXT_PUBLIC_MOCK_FLAG=True)
+def test_get_profile_data_from_gdpr_api(api_client):
```

Comment:
I'm going to add authentication tests when confirmed that it works in testing environment.